toolchain/check/facet_type.cpp

```diff
@@ -408,36 +408,45 @@ class SubstImplWitnessAccessCallbacks : public SubstInstCallbacks {
       SemIR::InstId inst_id;
     };
 
-    auto Find(Context& context, SemIR::ImplWitnessAccess access)
-        -> std::optional<Value> {
-      auto it = cache_.find(GetKey(context, access));
-      if (it == cache_.end()) {
+    // Finds and returns a pointer into the cache for a given ImplWitnessAccess.
+    // The pointer will be invalidated by mutating the cache.
+    auto FindRef(Context& context, SemIR::ImplWitnessAccess access)
+        -> std::optional<Value*> {
```

Comment:
In the toolchain we usually don't use `optional<T*>` and instead just use plain `T*` with a comment for optional pointers. ([Style guide](https://docs.carbon-lang.dev/docs/project/cpp_style_guide.html#:~:text=If%20the%20parameter%20is%20optional%2C%20use%20a%20pointer%20and%20document%20that%20it%20may%20be%20null.))